api/turing/api/routers_api.go

```diff
@@ -155,24 +160,32 @@ func (c RoutersController) UpdateRouter(r *http.Request, vars RequestVars, body
 	rVersion, err := request.BuildRouterVersion(
 		router, c.RouterDefaults, c.AppContext.CryptoService, c.AppContext.ExperimentsService, c.EnsemblersService)
 	if err == nil {
-		// Save router version, re-assign the value of err
+		// Save router version as undeployed if there is no need to deploy it
+		if !toDeploy {
+			rVersion.Status = models.RouterVersionStatusUndeployed
+		}
 		routerVersion, err = c.RouterVersionsService.Save(rVersion)
 	}
 
 	if err != nil {
-		return InternalServerError("unable to update router", err.Error())
+		if toDeploy {
+			return InternalServerError("unable to update router", err.Error())
+		}
+		return InternalServerError("unable to save router version", err.Error())
 	}
 
 	// Deploy the new version
-	go func() {
-		err := c.deployOrRollbackRouter(project, router, routerVersion)
-		if err != nil {
-			log.Errorf("Error deploying router %s:%s:%d: %v",
-				project.Name, router.Name, routerVersion.Version, err)
-		}
-	}()
-
-	return Ok(router)
+	if toDeploy {
+		go func() {
+			err := c.deployOrRollbackRouter(project, router, routerVersion)
```

Comment:
Creating a new router version now uses the same controller method used to update a router. The `toDeploy` flag, which is passed to the API as a URL query string, now determines which parts of the method to run, most notably the `deployOrRollbackRouter` call.